fail2ban/ipaddr.py

```diff
@@ -0,0 +1,294 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: t -*-
+# vi: set ft=python sts=4 ts=4 sw=4 noet :
+
+# This file is part of Fail2Ban.
+#
+# Fail2Ban is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# Fail2Ban is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Fail2Ban; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+# Author: Alexander Koeppe
+# 
+
+__author__ = "Alexander Koeppe"
+__copyright__ = "Copyright (c) 2016 Alexander Koeppe"
+__license__ = "GPL"
+
+import inspect
+import socket
+import struct
+import re
+
+from functools import wraps
+
+##
+# Helper functions / decorator
+# Thanks to Yaroslav Halchenko (yarikoptic)
+#
+def asip(ip):
+	"""A little helper to guarantee ip being an IPAddr instance"""
+	return ip if isinstance(ip, IPAddr) or ip == None else IPAddr(ip)
```

Comment:
BTW. That was me as a result of tests adaption, where "getBansMerged" is been called w/o any arguments, but the decorator has expected at least one madatory.